client/src/components/BulkCreateForm/BulkCreateForm.js

```diff
@@ -76,164 +172,298 @@ export default class BulkCreateForm extends React.Component {
+                            Our Fields
+                        </div>
+                    </div>
+                    {mapping.map(pair => (
+                        <div className="d-flex row" key={pair.originalField}>
+                            <div className="column d-flex align-items-center">{pair.originalField}</div>
+                            <div className="column d-flex align-items-center justify-content-center">
+                                <i className="material-icons">
+                                    arrow_right_alt
+                                </i>
+                            </div>
+                            <div className="column d-flex justify-content-end my-1">
+                                <Form.Control as="select" className="mr-3" value={pair.mappedField} onChange={event => {
+                                    pair.mappedField = event.currentTarget.value;
+                                    fields.forEach(field => {
+                                        field.selected = !!mapping.find(currentPair => {
+                                            return currentPair.mappedField === field.name;
+                                        });
+                                    });
+                                    this.setState({mapping, fields});
+                                }}>
+                                    <option value={null}>Select a Field</option>
+                                    {fields.map(field => (
+                                        <option key={field.name}
+                                                disabled={field.selected && field.name !== pair.mappedField}
+                                                value={field.name}>
+                                            {field.label}
+                                        </option>
+                                    ))}
+                                </Form.Control>
+                            </div>
+                        </div>
+                    ))}
+                </div>
+            </Card.Body>
+
+            <Card.Footer className="d-flex justify-content-end">
+                <Button variant="bare" onClick={() => this.resetForm(true)}>
+                    Cancel
+                </Button>
+                <Button onClick={(event) => this.submitForValidation(event)}>
+                    Continue
+                </Button>
+            </Card.Footer>
+        </>)
     }
 
-    /**
-     * Build the form object to send to the onSubmit handler
-     */
-    submitForm() {
+    renderValidationResults() {
         const { fileUpload } = this.state;
-        const { onCsvSubmit, onZipSubmit } = this.props;
-
-        let form = {
-            file: fileUpload.file
-        };
-
-        if (fileUpload.fileType === '.zip') {
-            onZipSubmit(form);
+        const { validationResult } = this.props;
+
+        // TODO: Handle general errors
```

Comment:
👀

Reply:
Yea basically I added an option to the csv validation where it can pass in general errors instead of line-based errors, but it's never used right now. I was expecting some use cases to come up where it might but they really didn't. The only thing I can think of right now is IOExceptions and maybe if it can't connect to S3 or something